### PR TITLE
LibWeb: Allow block level boxes to be floated and have clearance

### PR DESCRIPTION
**Pull request by kalenikaliaksandr**
Before, we completely ignored clearance for block-level boxes if they were floated. This was incorrect because it is valid for a block-level box to be floated and still have clearance. However, unlike clearance on normal flow boxes, clearance on floating boxes does not affect the y-position of subsequent normal flow boxes. Instead, it pushes the box's position to the very beginning of an edge.

### CODE REVIEW

File: Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp [hunk 1]
```diff
@@ -979,10 +980,18 @@ void BlockFormattingContext::layout_floating_box(Box const& box, BlockContainer
...
-            if (!did_touch_preceding_float || !did_place_next_to_preceding_float) {
+            auto has_clearance = false;
+            if (side == FloatSide::Left) {
+                has_clearance = computed_values.clear() == CSS::Clear::Left || computed_values.clear() == CSS::Clear::Both;
+            } else if (side == FloatSide::Right) {
+                has_clearance = computed_values.clear() == CSS::Clear::Right || computed_values.clear() == CSS::Clear::Both;
+            }
+
+            if (!did_touch_preceding_float || !did_place_next_to_preceding_float || has_clearance) {
                 // One of two things happened:
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp:991]: One of *three* things happened :sweat_smile:

**Review by awesomekling — CHANGES_REQUESTED** [review]
Missing "Fixes #21023" :^)

**Comment by kalenikaliaksandr** [comment]
> Missing "Fixes https://github.com/SerenityOS/serenity/issues/21023" :^)

Actually, there are more issues to fix for reduction in #21023 😅

**Comment by awesomekling** [comment]
> > Missing "Fixes #21023" :^)
> 
> Actually, there are more issues to fix for reduction in #21023 😅

Sure, then maybe just "Work towards #21023" is nice. So we can see where the test case originally came from :)

**Comment by kalenikaliaksandr** [comment]
> Sure, then maybe just "Work towards https://github.com/SerenityOS/serenity/issues/21023" is nice. So we can see where the test case originally came from :)

yes, sure, I added this in the commit description.

**Comment by awesomekling** [comment]
Sure okay :)

**Pull request merged**